assets/js/partners/contact.js

```diff
@@ -0,0 +1,113 @@
+  const form = document.getElementById('contact-us-form');
+  const formError = document.getElementById('form-error-message');
+  const formSuccess = document.getElementById('form-success-message');
+  const captchaError = document.getElementById('captcha-error-message');
+  const captcha = document.getElementById('g-recaptcha-response');
+
+  // skip captcha check when debug flag is turned on
+  if (!debug || (debug && !+debug.value)) {
+    // check for captcha response
+    if (!captcha || !captcha.value) {
+      captchaError.textContent = captchaError.dataset.error;
+      captchaError.classList.remove('display-none');
+      return false;
+    }
+  }
+
+  const honeypot = fetchTextInput('information');
+  if (honeypot) {
+    return false;
+  }
+
+  const firstName = fetchTextInput('first_name');
+  const lastName = fetchTextInput('last_name');
+  const jobTitle = fetchTextInput('job_title');
+  const email = fetchTextInput('email').toLowerCase();
+  const agency = fetchSelectInput('agency');
+  const program = fetchTextInput('program');
+  let launchDate = fetchTextInput('launch_date');
+
+  if (launchDate) {
+    launchDate = new Date(launchDate).toISOString().slice(0, 10);
+  }
+
+  const proofing = fetchSelectInput('proofing');
+  const estimatedUsers = fetchTextInput('estimated_users');
+  const estimatedAuths = fetchTextInput('estimated_auths');
+  const submittedDescription = fetchTextInput('description');
+  let description;
+
+  if (submittedDescription) {
+    description = submittedDescription;
+  } else {
+    description = 'I would like to learn more about Login.gov.';
+  }
+
+  const requestBody = {
+    request: {
+      requester: {
+        name: `${firstName} ${lastName}`,
+        email,
+      },
+      subject: `Login.gov inquiry for ${program}`,
+      comment: { body: description },
+      ticket_form_id: 5599398793108,
+      custom_fields: [
+        { id: 5549275172884, value: firstName },
+        { id: 5549276309140, value: lastName },
+        { id: 5549566619668, value: jobTitle },
+        { id: 4418412738836, value: agency },
+        { id: 5549623980820, value: program },
+        { id: 4417512374548, value: launchDate },
+        { id: 5549997170452, value: proofing },
+        { id: 4417514509076, value: estimatedUsers },
+        { id: 4417513940756, value: estimatedAuths },
+      ],
+    },
+  };
+
+  const req = new XMLHttpRequest();
+  const target = 'https://logingov.zendesk.com/api/v2/requests.json';
+  req.open('POST', target);
+  req.setRequestHeader('Content-Type', 'application/json');
+
+  req.onreadystatechange = () => {
+    if (req.readyState === XMLHttpRequest.DONE) {
+      if (req.status === 201) {
+        form.classList.add('display-none');
+        form.reset();
+        formSuccess.classList.remove('display-none');
+        return false;
+      }
+
+      formError.classList.remove('display-none');
+      return false;
+    }
+  };
+
+  req.send(JSON.stringify(requestBody));
```

Comment:
I expect the [Fetch API](https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API) could simplify this a fair bit. XMLHttpRequest is pretty clunky to work with.

```ts
async function submitPartnerContact(event) {
  // ...

  const response = await fetch('https://logingov.zendesk.com/api/v2/requests.json', {
    method: 'POST',
    body: JSON.stringify(requestBody),
    headers: { 'Content-Type': 'application/json' },
  });

  if (response.status === 201) {
    form.classList.add('display-none');
    form.reset();
    formSuccess.classList.remove('display-none');
  } else {
    formError.classList.remove('display-none');
  }
}
```

Reply:
that is _much_ better. Thanks for helping guide my growth as a front-end engineer 😝